app/models/concerns/request_item.rb

```diff
@@ -25,7 +39,44 @@ def duke_owned?
     record_owner == 'duke'
   end
 
+  def trln_partner_owned?
+    record_owner.in?(%w[unc ncsu nccu])
+  end
+
   def physical_item?
     fetch(TrlnArgon::Fields::ACCESS_TYPE, []).include?('At the Library')
   end
+
+  # These values are consistent, no matter the item or the home library's
+  # ILLiad instance
+  def illiad_request_params_fixed
+    {
+      'Value': 'GenericRequestTRLNLoan',
+      'genre': 'TRLNbook',
+      'RequestType': 'null',
+      'form': '20',
+      'Action': '10',
+      'IlliadForm': 'Logon',
+      'CitedPages': 'TRLN'
+    }
+  end
+
+  def illiad_request_params_variable
+    {
+      'Location': item_url_absolute,
+      'LoanTitle': fetch(TrlnArgon::Fields::TITLE_MAIN),
+      'ESPNumber': oclc_number,
+      'ISSN': isbn_number.join(' / '),
+      'LoanEdition': edition,
+      'LoanPublisher': imprint_main_for_header_display,
+      'LoanAuthor': statement_of_responsibility
+    }
+  end
+
+  def item_url_absolute
+    [
+      TrlnArgon::Engine.configuration.root_url.chomp('/'),
+      Rails.application.routes.url_helpers.solr_document_path(self)
```

Comment:
I think you can use `trln_solr_document_url` and get the complete URL without using `TrlnArgon::Engine.configuration.root_url` because this is being called in a controller context.

Reply:
@corylown Thanks -- I would love to use an existing method to get the item URL... just wasn't sure where to look. I get this error when I try`trln_solr_document_url` though:
undefined local variable or method `trln_solr_document_url' for #<SolrDocument:0x00000005a0b068>

Reply:
Unrelated -- but it also might be worth updating to the latest version of Argon as part of this PR.